packages/tools/src/types/EventTypes.ts

```diff
@@ -118,6 +118,10 @@ type AnnotationAddedEventDetail = {
  * drawing on the viewport.
  */
 type AnnotationCompletedEventDetail = {
+  /** unique id of the viewport */
+  viewportId: string;
+  /** unique id of the rendering engine */
+  renderingEngineId: string;
```

Comment:
I actually don't like this. The annotation completed event shouldn't include information about the viewport/renderingEngineIds. Can you make your way around by using some utility? like with `packages/tools/src/utilities/getViewportForAnnotation.ts`

Reply:
I had never used `getViewportForAnnotation`. That's good to know!
Updated